### PR TITLE
Add digest: Preparing for a coding interview

### PR DESCRIPTION
**Pull request by li-kai**
As per title. Much thanks to @yangshun!

### CODE REVIEW

**Comment by li-kai** [comment]
Deploy preview ready!

Built with commit 4d8d3e7c67210435cea010a9d2392011ebdf7b31

https://deploy-preview-104--nushackers.netlify.com

File: content/post/2017-10-05-digest-preparing-for-a-coding-interview.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+*This post was originally published at [https://medium.com/@yangshun](https://medium.freecodecamp.org/coding-interviews-for-dummies-5e048933b82b).*
+
+Yang Shun is an astute NUS alumni and co-creator of NUSMods, among many other things. He is current working at Facebook.
```
Comment (advaypal) [inline content/post/2017-10-05-digest-preparing-for-a-coding-interview.md]: currently

File: content/post/2017-10-05-digest-preparing-for-a-coding-interview.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+- Digest
+date: 2017-10-05T00:00:00Z
+published: true
+title: 'Digest: Preparing for a Coding Interview'
+url: /2017/09/17/digest-preparing-for-a-coding-interview/
+---
+
+<img src="/img/2017/10/android.jpeg" alt="android campus"/>
+
+Despite scoring decent grades in both my CS101 Algorithm class and my Data Structures class in university, I shudder at the thought of going through a coding interview that focuses on algorithms.
```
Comment (yangshun) [inline content/post/2017-10-05-digest-preparing-for-a-coding-interview.md]: For NUS context can just say CS1010/1020

File: content/post/2017-10-05-digest-preparing-for-a-coding-interview.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+{{< tweet 716997357624287232 >}}
+
+Front end is a specialized domain where engineers have to care about many issues related to browser compatibilities, the Document Object Model, JavaScript performance, CSS layouts, and so on. It is uncommon for front-end engineers to implement some of the complex algorithms tested in interviews.
+At companies like Facebook and Google, the people are software engineers first, domain experts second.
+
+Unfortunately, the rules are set by the companies, not by the candidates. There is a high emphasis on general computer science concepts like algorithms, design patterns, data structures; core skills that a good software engineer should possess. If you want the job, you have to play by the rules set by the game masters — improve your coding interview skills!
+
+<img src="/img/2017/10/handbook.png" alt="tech interview handbook"/>
```
Comment (yangshun) [inline content/post/2017-10-05-digest-preparing-for-a-coding-interview.md]: Why not using Markdown image syntax?

File: static/scss/main.scss [hunk 4]
```diff
@@ -60,6 +60,10 @@ body {
   margin: 0 auto;
 }
 
+.twitter-tweet {
+  margin: 0 auto;
```
Comment (yangshun) [inline static/scss/main.scss]: This seems more fitting to be a utility class, such as `.centralize` or something.

File: content/post/2017-10-05-digest-preparing-for-a-coding-interview.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+Another platform that allows you to practice coding interviews is [Pramp](https://pramp.com/). Where interviewing.io matches potential job seekers with seasoned technical interviewers, Pramp differs takes a different approach. Pramp pairs you up with another peer who is also a job seeker and both of you take turns to assume the role of interviewer and interviewee. Pramp also prepares questions for you, along with suggested solutions and prompts to guide the interviewee. Personally, I do not really like Pramp's approach because if I were to interview someone, I would rather choose a question I am familiar with. Also, many users of the platform do not have the experience of being interviewers and that can result in a horrible interview experience. There was once where my matched peer, as the interviewer, did not have the right understanding of the question and attempted to lead me down the wrong path of solving the question.
```
Comment (yangshun) [inline content/post/2017-10-05-digest-preparing-for-a-coding-interview.md]: An ending like this is quite abrupt 😂 

I recommend using the ending from the Medium article:

### Conclusion

Coding interviews are tough. But fortunately, you can get better at them by studying and practicing for them, and doing mock interviews.

To recap, to do well in coding interviews:

1. Decide on a programming language
1. Study CS fundamentals
1. Practice solving algorithm questions
1. Internalize the Do’s and Don’ts of what to do during interviews
1. Practice doing mock interviews
1. Interview successfully to get the job

By following these steps, you will improve your coding interview skills, and be one step closer (or probably more) to landing your dream job.

All the best!

File: content/post/2017-10-05-digest-preparing-for-a-coding-interview.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+I have used interviewing.io both as an interviewer and an interviewee and found the experience to be really great! [Aline Lerner](https://twitter.com/alinelernerLLC), the CEO and co-founder of interviewing.io and her team are passionate about revolutionizing the technical interview process and helping candidates to improve their skills at interviewing. She has also published a number of technical interview-related articles on the [interviewing.io blog](http://blog.interviewing.io/). interviewing.io is still in beta now but I recommend signing up as early as possible to increase the likelihood of getting an invite.
+
```
Comment (yangshun) [inline content/post/2017-10-05-digest-preparing-for-a-coding-interview.md]: Can we add the interviewing.io graphic here as well:

Image

File: content/post/2017-10-05-digest-preparing-for-a-coding-interview.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+Another platform that allows you to practice coding interviews is [Pramp](https://pramp.com/). Where interviewing.io matches potential job seekers with seasoned technical interviewers, Pramp differs takes a different approach. Pramp pairs you up with another peer who is also a job seeker and both of you take turns to assume the role of interviewer and interviewee. Pramp also prepares questions for you, along with suggested solutions and prompts to guide the interviewee. Personally, I do not really like Pramp's approach because if I were to interview someone, I would rather choose a question I am familiar with. Also, many users of the platform do not have the experience of being interviewers and that can result in a horrible interview experience. There was once where my matched peer, as the interviewer, did not have the right understanding of the question and attempted to lead me down the wrong path of solving the question.
+
+*This post was originally published at [https://medium.com/@yangshun](https://medium.freecodecamp.org/coding-interviews-for-dummies-5e048933b82b).*
```
Comment (yangshun) [inline content/post/2017-10-05-digest-preparing-for-a-coding-interview.md]: `This post was originally published on [Medium](https://medium.freecodecamp.org/coding-interviews-for-dummies-5e048933b82b).`

File: content/post/2017-10-05-digest-preparing-for-a-coding-interview.md (hunk 1 above)
Comment (yangshun) [reply content/post/2017-10-05-digest-preparing-for-a-coding-interview.md]: Thanks Advay for catching it! Can use this instead:

`He is currently working as a Front End Engineer at Facebook.`

File: static/scss/main.scss (hunk 4 above)
Comment (li-kai) [reply static/scss/main.scss]: That class is added by twitter 😅

**Pull request merged**